Solution.vb

```diff
@@ -0,0 +1,47 @@
+﻿Imports System.IO
+Module Module1
+
+    Dim searchRoot As String
+    Dim ext As String
+    Dim subStr As String
+
+
+    Public Function GetAllSuitableFilePaths(searchRoot, ext, subStr) As String()
+
+        If searchRoot = "" Then
+            Console.WriteLine("Hodnota adresáře nemůže být prázdná")
+            Exit Function
+        ElseIf System.IO.Directory.Exists(searchRoot) = False Then
+            Console.WriteLine("Uvedený adresář neexistuje")
+            Exit Function
+        ElseIf ext = "" Then
+            Console.WriteLine("Hodnota přípony nemůže být prázdná")
+            Exit Function
+        End If
+
+        Dim Directory As New IO.DirectoryInfo(searchRoot)
+        Dim FileArray As IO.FileInfo() = Directory.GetFiles("*" & subStr & "*" & ext, SearchOption.AllDirectories)
+        Dim File As IO.FileInfo
+
+
+        For Each File In FileArray
+            Console.WriteLine("File Full Name: {0}", File.FullName)

```

Comment:
Měla by to být funkce, která vrací ty názvy jako array, ne je vypsat do konzole. Není to nic velkého, ale bylo by fajn udělat ještě ten převod na String() jak je ta funkce předepsaná.